src/main/java/com/flipkart/yak/config/CompactionContext.java

```diff
@@ -56,10 +56,14 @@ public int hashCode() {
 
     @Override
     public void validate() throws ConfigurationException {
-        if (tableName==null && nameSpace == null && rsGroup == null) {
+        validateTable(tableName, nameSpace, rsGroup);
+    }
+
+    static void validateTable(String tableName, String nameSpace, String rsGroup) throws ConfigurationException {
+        if (tableName == null && nameSpace == null && rsGroup == null) {
```

Comment:
whats the requirement for this ?

Reply:
Its being shared with PromptCompactionRequest Class as well , hence decided to share it between the two classes through functions